pkg/prowgen/podspec_test.go

```diff
@@ -0,0 +1,414 @@
+		Branch:  "branch",
+		Variant: "variant",
+	}
+	tests := []struct {
+		name string
+	}{
+		{
+			name: "secret is added",
+		},
+	}
+	for _, tc := range tests {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+			g := NewCiOperatorPodSpecGenerator(meta)
+			g.Add(Claims()...)
+			podspec, err := g.Build()
+			if err != nil {
+				t.Fatalf("Unexpected error: %v", err)
+			}
+			testhelper.CompareWithFixture(t, podspec)
+		})
+	}
+}
+
+func TestLeaseClient(t *testing.T) {
+	t.Parallel()
+	meta := api.Metadata{
+		Org:     "org",
+		Repo:    "repository",
+		Branch:  "branch",
+		Variant: "variant",
+	}
+	tests := []struct {
+		name string
+	}{
+		{
+			name: "secret is added",
+		},
+	}
+	for _, tc := range tests {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+			podspec, err := NewCiOperatorPodSpecGenerator(meta).LeaseClient().Build()
+			if err != nil {
+				t.Fatalf("Unexpected error: %v", err)
+			}
+			testhelper.CompareWithFixture(t, podspec)
+		})
+	}
+}
+
+func TestSecrets(t *testing.T) {
+	t.Parallel()
+	meta := api.Metadata{
+		Org:     "org",
+		Repo:    "repository",
+		Branch:  "branch",
+		Variant: "variant",
+	}
+	tests := []struct {
+		name    string
+		secrets []*api.Secret
+	}{
+		{
+			name: "empty list is a nop",
+		},
+		{
+			name:    "one secret",
+			secrets: []*api.Secret{{Name: "sekret"}},
+		},
+		{
+			name:    "multiple secrets",
+			secrets: []*api.Secret{{Name: "sekret"}, {Name: "another", MountPath: "/with/path"}},
+		},
+	}
+	for _, tc := range tests {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+			g := NewCiOperatorPodSpecGenerator(meta)
+			g.Add(Secrets(tc.secrets...)...)
+			podspec, err := g.Build()
+			if err != nil {
+				t.Fatalf("Unexpected error: %v", err)
+			}
+			testhelper.CompareWithFixture(t, podspec)
+		})
+	}
+}
+
+func TestPromotion(t *testing.T) {
+	t.Parallel()
+	meta := api.Metadata{
+		Org:     "org",
+		Repo:    "repository",
+		Branch:  "branch",
+		Variant: "variant",
+	}
+	tests := []struct {
+		name string
+	}{
+		{
+			name: "secret and parameters are added",
+		},
+	}
+	for _, tc := range tests {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+			podspec, err := NewCiOperatorPodSpecGenerator(meta).Promotion().Build()
+			if err != nil {
+				t.Fatalf("Unexpected error: %v", err)
+			}
+			testhelper.CompareWithFixture(t, podspec)
+		})
+	}
+}
+
+func TestClusterProfile(t *testing.T) {
+	t.Parallel()
+	meta := api.Metadata{
+		Org:    "org",
+		Repo:   "repository",
+		Branch: "branch",
+	}
+	tests := api.ClusterProfiles()
+	for _, tc := range tests {
+		tc := tc
+		t.Run(string(tc), func(t *testing.T) {
+			t.Parallel()
+			podspec, err := NewCiOperatorPodSpecGenerator(meta).ClusterProfile(tc).Targets("test-name").Build()
+			if err != nil {
+				t.Fatalf("Unexpected error: %v", err)
+			}
+			testhelper.CompareWithFixture(t, podspec)
+		})
+	}
+
+	errorTests := []struct {
```

Comment:
nit: which errors in this file cannot be described with this three-member struct? What's the utility of copy-pasting the test setup and parallel execution logic in so many individual tests instead of just using subtests in one func?